plugins/invisible-typing/src/index.ts

```diff
@@ -28,7 +28,7 @@ export async function start(): Promise<void> {
       const globalInvisible = cfg.get("invisible", true);
       const channelWise = cfg.get("button", true) ? cfg.get("channelWise", true) : false;
       const channels = cfg.get("channels", { [channelId]: globalInvisible });
-      if (channelWise ? channels[channelId] : globalInvisible) {
+      if (channelWise ? channels[channelId] ?? globalInvisible : globalInvisible) {
```

Comment:
<a href="https://sourcery.ai"><img src="https://sourcery.ai/favicon.svg" height="15" alt="Sourcery logo" /></a>  **suggestion (code_refinement):** Consistent use of nullish coalescing for channel visibility checks.

This ensures that the logic for determining whether typing should be invisible is consistent across the plugin, using the global setting as a fallback.

```suggestion
if ((channelWise ? channels[channelId] : null) ?? globalInvisible) {
```